Utilities/XrdAdaptor/src/XrdRequestManager.cc

```diff
@@ -107,6 +109,8 @@ RequestManager::RequestManager(const std::string &filename, XrdCl::OpenFlags::Fl
       m_distribution(0,100),
       m_excluded_active_count(0)
 {
+    edm::Service<XrdStatisticsService> statsService;
+    if (statsService.isAvailable()) {m_stats = &(*statsService);}
```

Comment:
Since you are worried about service availability on a given thread maybe you should instead
-create a singleton
-have the XrdStaticstsService read from that singleton (or initialize the singleton as well?)
It appears the only reason you created a service was to listen to the 'endJob' transition.